### DISCUSSION TITLE
How do I reference a dll?

### DISCUSSION THREAD

**Discussion by EsatMan** [stride3d/stride#2356] [Q&A] [answered]
I referenced a dll in VS, it seems fine when i was coding, but I got an error said "system can't find the file" when I tried to run the game. And then I draged that dll into Stride, it became a .sdraw file and I still can't reference it yet. Is nuget the only way to reference code？

**Comment by Doprez** [comment]
This is likely a .NET setting instead of being something in Stride specific. In VS there should be a properties section of the file, can you make sure the DLL is being added to the bin on build? 

something like the below may work:
```
<ItemGroup>
    <Content Include="libs\a.dll" >
        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
    <Content Include="libs\b.dll" >
        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
</ItemGroup>
```
https://stackoverflow.com/questions/1776060/how-to-make-visual-studio-copy-a-dll-file-to-the-output-directory

**Reply by EsatMan** [reply, marked as answer]
Thanks for reply, I found the solution after a little test. I only referenced the dll in XXX.Game project but not XXX.Windows project, and when I copied the definitions to the .csproj of XXX.Windows, the game runs normally. It's a bit confusing that I had to deal dependencies manually in every project but it will do the job.
